### PR TITLE
chore: update ts-eslint to v2

### PR DESCRIPTION
**Pull request by emmenko**
Breaking changes release notes: https://github.com/typescript-eslint/typescript-eslint/releases/tag/v2.0.0

NOTE: the main question I have is about enabling the `@typescript-eslint/recommended-requiring-type-checking`, as it "slows down" the linting process as it needs to perform some typechecking as well.
From their docs they mention this:

image

I'm not sure how "splitting into stages" works though...

We can also keep the extra config disabled for now.

### CODE REVIEW

File: .eslintrc.yaml [hunk 1]
```diff
@@ -4,8 +4,11 @@ parserOptions:
     jsx: true
   ecmaVersion: 8
   sourceType: module
+  project: './tsconfig.eslint.json'
```
Comment (tdeekens) [inline .eslintrc.yaml]: Yeah did the same in flopflip last weekend :)

File: packages/application-shell/src/components/quick-access/butler/butler.js [hunk 2]
```diff
@@ -249,37 +249,34 @@ export class Butler extends React.Component {
...
-                      }
-                    : null
-                );
-              } else {
-                this.shake();
-              }
-            },
-            () => this.setNetworkError()
-          );
+          const nextCommands = this.props.getNextCommands(command);
```
Comment (tdeekens) [inline packages/application-shell/src/components/quick-access/butler/butler.js]: This is not sync, is it?

File: packages/mc-scripts/extract-intl.js [hunk 3]
```diff
@@ -225,32 +225,30 @@ const extractFromFile = async fileName => {
   // We usually do not build the translation-files because they are
   // handled by transifex and sync by the transifex CLI.
   if (shouldBuildTranslations) {
-    await Promise.all(
-      locales.map(async locale => {
-        const translationFileName = `${outputPath}/${locale}.json`;
+    locales.forEach(locale => {
```
Comment (tdeekens) [inline packages/mc-scripts/extract-intl.js]: Ah so this was wrapping it into a promise needlessly?

**Comment by emmenko** [comment]
Yeah well, for example when commiting, the lint-staged runs and takes much longer now.

File: packages/mc-scripts/extract-intl.js (hunk 3 above)
Comment (emmenko) [reply packages/mc-scripts/extract-intl.js]: Yes, that was one of the new rules from the additional config

Comment (tdeekens) [reply packages/mc-scripts/extract-intl.js]: I like it. Async behaviour in JS is "toxic"/breaking to the consumer so should be avoided.

File: packages/application-shell/src/components/quick-access/butler/butler.js (hunk 2 above)
Comment (emmenko) [reply packages/application-shell/src/components/quick-access/butler/butler.js]: Yes it's sync. The commands return a plain object or a list of objects. Never a promise.

**Comment by tdeekens** [comment]
We could also "split" the usage of the new rules and the update to v2.

**Comment by emmenko** [comment]
What do you mean? The new rules are part of v2 as far as I understand...

**Comment by tdeekens** [comment]
Split out the `@typescript-eslint/recommended-requiring-type-checking` into another PR from the upgrade to v2.

**Comment by emmenko** [comment]
What would be the benefit of doing that? 🤔

**Comment by tdeekens** [comment]
I thought that slows the linting down. If it doesn't I got it wrong.

**Comment by emmenko** [comment]
Well yes it's slower now, I just don't understand what splitting it into a separate PR would solve. I would say we try that and if it's too painful we can turn it off...

**Pull request merged**